presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -547,6 +551,27 @@ public static Double lineLocatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlic
         return new LengthIndexedLine(line).indexOf(point.getCoordinate()) / line.getLength();
     }
 
+    @Description("Returns the point in the line at the fractional length.")
+    @ScalarFunction("line_interpolate_point")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice lineInterpolatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlice, @SqlType(DOUBLE) double fraction)
+    {
+        if (!(0.0 <= fraction && fraction <= 1.0)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Fraction must be between 0 and 1, but is %s", fraction));
```

Comment:
Consider including function name in the error message. Otherwise, it might be challenging for the user to figure out which part of a complex query is causing this error.